### PR TITLE
Generate changelogs for 3.10.3.0

### PR DESCRIPTION
**Pull request by Kleidukos**
Cabal 3.10.3.0 Release process: Changelogs generation.

This PR needs to be ported to `master`.

Reviewers please make sure no changelog entry is already present in a previous release.

### CODE REVIEW

**Comment by Mikolaj** [comment]
O, here is Assorted/Changelogs! So it was still running at the time of 3.10 fork off. I think it rightly requires that the changelog config is retained even when changelogs are deleted.

**Review by Mikolaj — APPROVED** [review]
Thank you. 

Sanity check: it mentions https://github.com/haskell/cabal/pull/9459 that is the most crucial fix of the release AFAIR.

**Review by ulysses4ever — APPROVED** [review]
Full disclosure: I didn't check the entries from the new changelog with the ones in the old changelogs.

I also think more backticks could be employed to highlight technical entities (e.g. `extra-prog-path`). But we are unpaid volunteers, not someone paid by a backtick, so…

File: release-notes/Cabal-3.10.3.0.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Cabal and Cabal-syntax 3.10.3.0 changelog and release notes
+---
+
+## Release 3.10.3.0 is strictly a bug-fix release, with the fixes listed below
```
[the remark this answers is not in the document]
Comment (Kleidukos) [reply release-notes/Cabal-3.10.3.0.md:4]: I'm trying to stay consistent with previous iterations: https://github.com/haskell/cabal/blob/master/release-notes/Cabal-3.10.2.0.md and https://github.com/haskell/cabal/blob/master/release-notes/Cabal-3.10.2.1.md. You are right it doesn't make sense semantically speaking, but I'd rather change the files in another PR.

[the remark this answers is not in the document]
Comment (ulysses4ever) [reply release-notes/Cabal-3.10.3.0.md:4]: I did checkout older files before commenting. Only 3.10 seems to be doing it. (Some others I looked at just don't have the phrase at all though.) But that's okay.

[the remark this answers is not in the document]
Comment (Kleidukos) [reply release-notes/Cabal-3.10.3.0.md:4]: I'm all for changing this later. :)

**Review by ffaf1 — APPROVED** [review]

**Comment by Kleidukos** [comment]
@mergifyIO backport master

**Pull request merged**